freedomofpress/securedrop
# Updated testinfra tests to optionally run against a prod instance

## Status

Ready for Review

## Description of Changes

Fixes #4216 

Updates testinfra tests to allow them to be run against a production SecureDrop hardware instance from an Admin Workstation. 

## Testing

### Verify staging unchanged:
- check out this branch and run:
  ```
  make build-debs && make staging && make testinfra
  ```
  - [ ] testinfra tests complete without error

### test against prod hardware (assuming default IPs, DNS settings, and both v2 and v3 services enabled):
- on the admin workstation for a previously configured instance, `cd ~/Persistent/securedrop` and check out this branch
- run `./securedrop-admin setup -t` to install `testinfra` dependencies
- run `./securedrop-admin verify`
  -  [ ] testinfra tests complete with no errors
- remove the test dependencies with `rm -rf ~/Persistent/securedrop/admin/.venv3`
- run `./securedrop-admin setup` to install vanilla dependencies
- verify `testinfra` module not installed with:

  ```
  cd ~/Persistent/securedrop
  source admin/.venv3/bin/activate
  pip list | grep testinfra  # there should be no output
  ```
  
## Deployment

 No special requirements for deployment

## Checklist

### because securedrop-admin changes:

- `make -C admin test` passing locally.

### If you made non-trivial code changes:
